### PR TITLE
refactor: llmq only logging

### PR DESCRIPTION
**Pull request by ogabrielides**
## What was done?

Enabled a particularly long logging only if llmq logging category is active 

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by PastaPastaPasta — DISMISSED** [review]
utACK for squash merge. LGTM

File: src/llmq/dkgsession.cpp [hunk 1]
```diff
@@ -88,11 +88,14 @@ bool CDKGSession::Init(const CBlockIndex* _pQuorumBaseBlockIndex, const std::vec
     if (utils::IsQuorumRotationEnabled(params.type, m_quorum_base_block_index)) {
         int cycleQuorumBaseHeight = m_quorum_base_block_index->nHeight - quorumIndex;
         const CBlockIndex* pCycleQuorumBaseBlockIndex = m_quorum_base_block_index->GetAncestor(cycleQuorumBaseHeight);
-        std::stringstream ss;
-        for (const auto& mn : members) {
-            ss << mn->dmn->proTxHash.ToString().substr(0, 4) << " | ";
+        if (LogAcceptCategory(BCLog::LLMQ)) {
+            std::stringstream ss;
+            for (const auto &mn: members) {
```
Comment (PastaPastaPasta) [inline src/llmq/dkgsession.cpp:93]: Shouldn't change the orientation (keep the ref left)

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

File: src/llmq/dkgsession.cpp (hunk 1 above)
Comment (PastaPastaPasta) [reply src/llmq/dkgsession.cpp:93]:
```suggestion
            for (const auto& mn: members) {
```

File: src/llmq/dkgsession.cpp [hunk 2]
```diff
@@ -103,11 +106,13 @@ bool CDKGSession::Init(const CBlockIndex* _pQuorumBaseBlockIndex, const std::vec
     if (!myProTxHash.IsNull()) {
         dkgDebugManager.InitLocalSessionStatus(params, quorumIndex, m_quorum_base_block_index->GetBlockHash(), m_quorum_base_block_index->nHeight);
         relayMembers = utils::GetQuorumRelayMembers(params, m_quorum_base_block_index, myProTxHash, true);
-        std::stringstream ss;
-        for (const auto& r : relayMembers) {
-            ss << r.ToString().substr(0, 4) << " | ";
+        if (LogAcceptCategory(BCLog::LLMQ)) {
+            std::stringstream ss;
+            for (const auto &r: relayMembers) {
```
Comment (PastaPastaPasta) [inline src/llmq/dkgsession.cpp:111]:
```suggestion
            for (const auto& r: relayMembers) {
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Comment by ogabrielides** [comment]
@UdjinM6 Applied suggestions

File: src/llmq/dkgsession.cpp [hunk 3]
```diff
@@ -85,14 +85,14 @@ bool CDKGSession::Init(const CBlockIndex* _pQuorumBaseBlockIndex, const std::vec
...
+    if (LogAcceptCategory(BCLog::LLMQ) && utils::IsQuorumRotationEnabled(params.type, m_quorum_base_block_index)) {
         int cycleQuorumBaseHeight = m_quorum_base_block_index->nHeight - quorumIndex;
         const CBlockIndex* pCycleQuorumBaseBlockIndex = m_quorum_base_block_index->GetAncestor(cycleQuorumBaseHeight);
         std::stringstream ss;
-        for (const auto& mn : members) {
+        for (const auto& mn: members) {
             ss << mn->dmn->proTxHash.ToString().substr(0, 4) << " | ";
         }
-        LogPrintf("DKGComposition h[%d] i[%d] DKG:%s\n", pCycleQuorumBaseBlockIndex->nHeight, quorumIndex, ss.str());
+        logger.Batch("DKGComposition h[%d] i[%d] DKG:%s\n", pCycleQuorumBaseBlockIndex->nHeight, quorumIndex, ss.str());
```
Comment (UdjinM6) [inline src/llmq/dkgsession.cpp:95]: you missed this one
```suggestion
        logger.Batch("DKGComposition h[%d] i[%d] DKG:%s", pCycleQuorumBaseBlockIndex->nHeight, quorumIndex, ss.str());
```

**Comment by ogabrielides** [comment]
@UdjinM6 Done

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

File: src/llmq/dkgsession.cpp [hunk 4]
```diff
@@ -85,14 +85,14 @@ bool CDKGSession::Init(const CBlockIndex* _pQuorumBaseBlockIndex, const std::vec
 
     CDKGLogger logger(*this, __func__);
 
-    if (utils::IsQuorumRotationEnabled(params.type, m_quorum_base_block_index)) {
+    if (LogAcceptCategory(BCLog::LLMQ) && utils::IsQuorumRotationEnabled(params.type, m_quorum_base_block_index)) {
         int cycleQuorumBaseHeight = m_quorum_base_block_index->nHeight - quorumIndex;
         const CBlockIndex* pCycleQuorumBaseBlockIndex = m_quorum_base_block_index->GetAncestor(cycleQuorumBaseHeight);
         std::stringstream ss;
-        for (const auto& mn : members) {
+        for (const auto& mn: members) {
```
Comment (PastaPastaPasta) [inline src/llmq/dkgsession.cpp:92]:
```suggestion
        for (const auto& mn : members) {
```

File: src/llmq/dkgsession.cpp [hunk 5]
```diff
@@ -103,11 +103,13 @@ bool CDKGSession::Init(const CBlockIndex* _pQuorumBaseBlockIndex, const std::vec
     if (!myProTxHash.IsNull()) {
         dkgDebugManager.InitLocalSessionStatus(params, quorumIndex, m_quorum_base_block_index->GetBlockHash(), m_quorum_base_block_index->nHeight);
         relayMembers = utils::GetQuorumRelayMembers(params, m_quorum_base_block_index, myProTxHash, true);
-        std::stringstream ss;
-        for (const auto& r : relayMembers) {
-            ss << r.ToString().substr(0, 4) << " | ";
+        if (LogAcceptCategory(BCLog::LLMQ)) {
+            std::stringstream ss;
+            for (const auto& r: relayMembers) {
```
Comment (PastaPastaPasta) [inline src/llmq/dkgsession.cpp:108]:
```suggestion
            for (const auto& r : relayMembers) {
```

**Review by PastaPastaPasta — COMMENTED** [review]
Not sure why we changed this

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**